src/app/tabs/camera.cpp

```diff
@@ -0,0 +1,143 @@
+#include <app/config.hpp>
+#include <app/tabs/camera.hpp>
+#include <app/window.hpp>
+#include <app/theme.hpp>
+
+CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
+{
+    QStackedLayout *stack = new QStackedLayout(this);
+    stack->addWidget(connect_widget());
+    stack->addWidget(cam_widget());
+
+    connect(this, &CameraTab::stream_connected, this, [stack]() {stack->setCurrentIndex(1);} );
+    connect(this, &CameraTab::stream_disconnected, this, [stack]() {stack->setCurrentIndex(0);} );
+
+    player = NULL;
+    // connect_stream(); TODO: offer auto-connect option
+}
+
+QWidget* CameraTab::cam_widget()
+{
+    QWidget* widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+
+    QHBoxLayout *connection = new QHBoxLayout;
+    status = new QLabel("Disconnected", widget);
+    status->setFont(Theme::font_16);
+    QPushButton *disconnect = new QPushButton("disconnect");
+    disconnect->setFont(Theme::font_14);
+    disconnect->setFlat(true);
+    disconnect->setIconSize(Theme::icon_36);
+    Theme::get_instance()->add_button_icon("wifi", disconnect);
+    connection->addWidget(status);
+    connection->addStretch();
+    connection->addWidget(disconnect);
+
+    connect(disconnect, &QPushButton::clicked, this, &CameraTab::disconnect_stream);
+
+    video_widget = new QVideoWidget;
+
+    layout->addWidget(video_widget);
+    layout->addLayout(connection);
+
+    return widget;
+}
+
+

```

Comment:
remove extra newline (did you try sneaking this in? :smiley:)

Reply:
Yes

(No)